Detectors/MUON/MCH/Workflow/include/MCHWorkflow/DataDecoder.h

```diff
@@ -0,0 +1,132 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    DatDecoder.cxx
+/// \author  Andrea Ferrero
+///
+/// \brief Implementation of a data processor to run the raw decoding
+///
+
+#include <random>
+#include <iostream>
+#include <fstream>
+#include <stdexcept>
+#include "Framework/CallbackService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+#include "Framework/WorkflowSpec.h"
+
+#include "DPLUtils/DPLRawParser.h"
+#include "MCHBase/Digit.h"
+#include "Headers/RAWDataHeader.h"
+#include "Headers/RDHAny.h"
+#include "MCHRawCommon/DataFormats.h"
+#include "MCHRawDecoder/PageDecoder.h"
+#include "MCHRawElecMap/Mapper.h"
+//#include "MCHRawCommon/RDHManip.h"
+#include "MCHMappingInterface/Segmentation.h"
+#include "MCHWorkflow/DataDecoderSpec.h"
+//#include <array>
+
+#define MCH_MERGER_FEEID_MAX 63
+
+namespace o2::header
+{
+extern std::ostream& operator<<(std::ostream&, const o2::header::RAWDataHeaderV4&);
+}
+
+namespace o2
+{
+namespace mch
+{
+namespace raw
+{
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::mch::mapping;
+using RDHv4 = o2::header::RAWDataHeaderV4;
+
+class MergerBase;
+
+//_________________________________________________________________
+//
+// Data decoder
+//_________________________________________________________________
+class DataDecoder
+{
+ private:
+  void initElec2DetMapper(std::string filename);
+  void initFee2SolarMapper(std::string filename);
+
+ public:
```

Comment:
that's a recurring pattern I see in some workflows : why don't you provide a proper constructor of the DataDecoder class instead of having a default (and useless) one plus a bunch of setXXX methods ? Makes it more difficult to understand what's needed to get a working DataDecoder built IMO

Reply:
Done. All setXXX methods are replaced by a constructor.